nexus/reconfigurator/planning/tests/output/planner_expunge_clickhouse_zones_after_policy_is_changed_3_4.txt

```diff
@@ -121,31 +121,27 @@ to:   blueprint d895ef50-9978-454c-bdfb-b8dbe2c9a918
     -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
     dataset name                                                                                                     dataset uuid                           quota     reservation   compression
     -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
-    oxp_5fe54077-c016-49a9-becb-14993f133d43/crucible                                                                2a41b700-5494-4aa1-900b-20618125c1d0   none      none          off        
-    oxp_51c788ff-de33-43f7-b9c5-f5f56bf80736/crucible                                                                0bbf7ff9-f6a6-49d8-8dc4-c75b2fdb5531   none      none          off        
+    oxp_f52832ea-60d7-443b-9847-df5384bfc8e2/crucible                                                                cc43cb24-2e55-415d-9a2a-49a6b924b284   none      none          off        
+    oxp_9825ff38-f07d-44a1-9efc-55a25e72015b/crucible                                                                34de0f9d-ab3a-4047-bbf3-d2bf4beb2995   none      none          off        
```

Comment:
I tried to look at why this output changed. It looks like this list is sorted _only_ by kind; within one kind, there doesn't appear to be any sorting. Maybe we can make this more stable - if this analysis looks right, maybe this fixes it?

```diff
--- a/nexus/types/src/deployment/blueprint_diff.rs
+++ b/nexus/types/src/deployment/blueprint_diff.rs
@@ -562,7 +562,7 @@ impl BpTableData for DiffDatasetsDetails {
         // by sorting by dataset kind: after UUID redaction, that produces
         // a stable table ordering for datasets.
         let mut rows = self.datasets.values().collect::<Vec<_>>();
-        rows.sort_unstable_by_key(|d| &d.kind);
+        rows.sort_unstable_by_key(|d| (&d.kind, &d.name));
         rows.into_iter().map(move |dataset| {
             BpTableRow::from_strings(state, dataset.as_strings())
         })

Reply:
I think this is a good change. It's going to result in a bit of churn in other output files though. I'd suggest looking at the coming commit standalone.

Reply:
LGTM :+1: